core/opendaq/modulemanager/src/module_manager_impl.cpp

```diff
@@ -428,215 +428,68 @@ ErrCode ModuleManagerImpl::createDevice(IDevice** device, IString* connectionStr
-                {
-                    const auto newCap = replaceOldProtocolIds(cap);
-                    devInfoInternal.removeServerCapability(cap.getProtocolId());
-                    devInfoInternal.addServerCapability(newCap);
-                }
+                // automatically skips streaming connection for local and pseudo (streaming) devices
+                if (const auto mirroredDeviceConfigPtr = devicePtr.asPtrOrNull<IMirroredDeviceConfig>(true); mirroredDeviceConfigPtr.assigned())
+                    configureStreamings(mirroredDeviceConfigPtr, config);
             }
 
-            if (discoveredDeviceInfo.assigned())
-            {
-                // Replaces the default fields of capabilities retrieved from the config/structure protocol
-                // if a better alternative is available from the discovery results
-                for (const auto& capability : discoveredDeviceInfo.getServerCapabilities())
-                {
-                    const auto capId = capability.getProtocolId();
-                    ServerCapabilityPtr capPtr = capability;
-
-                    if (connectedDeviceInfo.hasServerCapability(capId))
-                    {
-                        try
-                        {
-                            capPtr = mergeDiscoveryAndDeviceCap(capability, connectedDeviceInfo.getServerCapability(capId));
-                        }
-                        catch ([[maybe_unused]] const std::exception& e)
-                        {
-                            capPtr = capability;
-                            LOG_W("{}: Failed to merge discovery and device server capability with ID {}: {}", module.getName(), capId, e.what())
-                        }
-
-                        connectedDeviceInfoInternal.removeServerCapability(capability.getProtocolId());
-                    }
-
-                    connectedDeviceInfoInternal.addServerCapability(capPtr);
-                }
-            }
-
-            const auto configConnectionInfo = connectedDeviceInfo.getConfigurationConnectionInfo();
-            if (configConnectionInfo.assigned() && configConnectionInfo.getProtocolType() != ProtocolType::Unknown)
-                completeServerCapabilities(configConnectionInfo, connectedDeviceInfo.getServerCapabilities());
-
-            // automatically skips streaming connection for local and pseudo (streaming) devices
-            auto mirroredDeviceConfigPtr = devicePtr.asPtrOrNull<IMirroredDeviceConfig>(true);
-            if (mirroredDeviceConfigPtr.assigned())
-            {
-                errCode = daqTry([this, &mirroredDeviceConfigPtr, &configPtr]
-                    {
-                        configureStreamings(mirroredDeviceConfigPtr, configPtr);
-                        return OPENDAQ_SUCCESS;
-                    });
-            }
+            return err;
         }
-
-        return errCode;
+    }
+    catch (const DaqException& e)
```

Comment:
the same as in the daqTry

Reply:
Yes, might just be a bit of a stylistic preference from my side, but I prefer actual try/catch statements when encasing such large code blocks.